### PR TITLE
Adding fortran wrapper for ex_put_partial_var

### PR DESCRIPTION
**Pull request by bourdin**
This is not ready for inclusion. @gsjaardema can you help me figure out why the wrapper expo does not show up in `libexoIIv2for.a`?

### CODE REVIEW

**Comment by bourdin** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by bourdin** [comment]
Duh, that is what exo_jack_32.c is for... Going back to work!

File: packages/seacas/libraries/exodus_for/src/addrwrap.F [hunk 1]
```diff
@@ -3950,7 +3950,52 @@ subroutine expecm (idexo, map_id, elem_ids, side_ids,
...
+        INTEGER*4 obj_id4       ! (R)
+        INTEGER*4 start_index4  ! (R)
+        INTEGER*4 num_entities4 ! (R)
+        INTEGER*4 ierr4         ! (W)
+
+		idexo4        = idexo
+		time_step4    = time_step
+		var_type4     = var_type
+		var_index4    = var_index
+		obj_id4       = obj_id
```
Comment (gsjaardema) [inline packages/seacas/libraries/exodus_for/src/addrwrap.F:3988]: Don't need `obj_id4`

File: packages/seacas/libraries/exodus_for/src/addrwrap.F [hunk 2]
```diff
@@ -3950,7 +3950,52 @@ subroutine expecm (idexo, map_id, elem_ids, side_ids,
...
+		start_index4  = start_index
+		num_entities4 = num_entities
+		ierr4         = ierr
+
+        CALL exppv4(idexo, time_step, var_type, var_index,
```
Comment (gsjaardema) [inline packages/seacas/libraries/exodus_for/src/addrwrap.F:3993]: This needs `CALL exppv4(idexo4, time_step4, var_type4, var_index4, obj_id, start_index, num_entities, var_vals, ierr4)`

File: packages/seacas/libraries/exodus_for/src/addrwrap.F [hunk 3]
```diff
@@ -3950,7 +3950,52 @@ subroutine expecm (idexo, map_id, elem_ids, side_ids,
...
+        INTEGER*4 num_entities4 ! (R)
+        INTEGER*4 ierr4         ! (W)
+
+		idexo4        = idexo
+		time_step4    = time_step
+		var_type4     = var_type
+		var_index4    = var_index
+		obj_id4       = obj_id
+		start_index4  = start_index
+		num_entities4 = num_entities
```
Comment (gsjaardema) [inline packages/seacas/libraries/exodus_for/src/addrwrap.F:3990]: `start_index` and `num_entities` are 64-bit integers in the C API, so don't need the `4` variants of these.

**Review by gsjaardema — COMMENTED** [review]
A few changes in the addrwrap.F needed...

**Comment by gsjaardema** [comment]
Yes, there is code duplication between `exodus_for` and `exoIIv2for32` ...

File: packages/seacas/libraries/exodus_for/src/addrwrap.F [hunk 4]
```diff
@@ -3950,7 +3950,146 @@ subroutine expecm (idexo, map_id, elem_ids, side_ids,
...
+        INTEGER*4 var_type4     ! (R)
+        INTEGER*4 var_index4    ! (R)
+        INTEGER*4 obj_id4       ! (R)
+        INTEGER*4 ierr4         ! (W)
+
+		idexo4        = idexo
+		time_step4    = time_step
+		var_type4     = var_type
+		var_index4    = var_index
+		obj_id4       = obj_id
```
Comment (gsjaardema) [inline packages/seacas/libraries/exodus_for/src/addrwrap.F:4044]: `obj_id` should be of type `ex_entity_id` which is an `int64_t`, so it should be passed directly through.

**Review by gsjaardema — CHANGES_REQUESTED** [review]
Looks good.  The only issue I saw was with the `obj_id`.  Thanks for doing this

**Review by gsjaardema — APPROVED** [review]

**Comment by gsjaardema** [comment]
Yes, I have thought about that a few times, but never had time to do or learn more about them.  Would definitely welcome a PR with that improvement...

I'm not sure how the 8-byte real and int compiler option that we use for fortran interacts with those; hopefully not an issue.

**Comment by gsjaardema** [comment]
Thanks.  Very much appreciated.

**Pull request merged**